### ISSUE TITLE
Clarify Supported Versions of Go

### ISSUE THREAD

**Issue by tri-adam** [sylabs/singularity#345]
I believe that we may want to clarify the supported versions of Go for SingularityCE. The [Go Release Maintenance Policy](https://github.com/golang/go/wiki/Go-Release-Cycle#release-maintenance) states that "minor releases to address security problems for Go 1.x stop once Go 1.x+2 is released."

Distributions do not always provide up to date Go toolchains, and the [SingularityCE installation documentation](https://github.com/sylabs/singularity/blob/master/INSTALL.md#install-go) recommends installing from upstream Go intead:

> Singularity is written in Go, and may require a newer version of Go than is available in the repositories of your distribution. We recommend installing the latest version of Go from the official binaries.

At the moment, we stop short of explicitly stating supported versions. This causes a few issues:

- It's possible that a user will build SingularityCE from source with a version of the Go toolchain that has a critical security issue, for which no security patches are available.
- It may force us to skip updates of external dependencies. For example, the `mvdan.cc/sh/v3` module has dropped support for Go 1.15 as of v3.4.0 ([ref](https://github.com/mvdan/sh/releases/tag/v3.4.0)). See #342 and #343 for real world examples of the implications of this.

Given the above, I suggest that we explicitly state that the supported Go language versions of SingularityCE mirror that of the Go language itself.

**Comment by dtrudg** [comment]
Just noting that at present our unstated but implied Go compatibility guarantee comes from our CI performing `rpmbuild` with EPEL's version of Go. This is ultimately the deciding version constraint that will have to be broken, or maintained here.

We have a community meeting this Thursday, which would be a good time to raise the issues here. Would you want to lead on that @tri-adam

**Comment by tri-adam** [comment]
> Just noting that at present our unstated but implied Go compatibility guarantee comes from our CI performing `rpmbuild` with EPEL's version of Go. This is ultimately the deciding version constraint that will have to be broken, or maintained here.

Yes, and also the [Building & Installing from an RPM](https://github.com/sylabs/singularity/blob/master/INSTALL.md#building--installing-from-an-rpm) section of `INSTALL.md`, as this currently installs the `golang` package from EPEL.

> We have a community meeting this Thursday, which would be a good time to raise the issues here. Would you want to lead on that @tri-adam ?

Great idea, I would be happy to. 

@DrDaveD, tagging you in case you wish to voice an opinion here (or on the community call!)
